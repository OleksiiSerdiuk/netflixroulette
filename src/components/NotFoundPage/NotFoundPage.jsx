import { Link } from 'react-router-dom';

const NotFound = () => {
  return (
    <div className='section-not-found-page'>
      <h1>404 Not Found</h1>
      <Link className='home-link' to='/'>
        Go to main page
      </Link>
    </div>
  );
};

export default NotFound;
