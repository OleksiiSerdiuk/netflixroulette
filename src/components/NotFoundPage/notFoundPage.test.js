import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';
import NotFoundPage from './NotFoundPage';

test('Shows the not found page message for a wrong movie page', () => {
  const history = createMemoryHistory();
  history.push('/movie-not-found');

  const { asFragment } = render(
    <Router location={history.location} navigator={history}>
      <NotFoundPage />
    </Router>
  );

  expect(screen.getByText(/404 Not Found/i)).toBeInTheDocument();
  expect(asFragment(<NotFoundPage />)).toMatchSnapshot();
});
