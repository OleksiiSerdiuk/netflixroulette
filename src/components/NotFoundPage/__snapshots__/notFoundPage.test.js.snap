// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shows the not found page message for a wrong movie page 1`] = `
<DocumentFragment>
  <div
    class="section-not-found-page"
  >
    <h1>
      404 Not Found
    </h1>
    <a
      class="home-link"
      href="/"
    >
      Go to main page
    </a>
  </div>
</DocumentFragment>
`;
